internal/controller/appconfigurationprovider_controller.go

```diff
@@ -425,36 +425,43 @@ func (reconciler *AzureAppConfigurationProviderReconciler) createOrUpdateSecrets
 		Namespace: provider.Namespace,
 	}
 
-	for secretName, secret := range settings.SecretSettings {
-		secretObj := &corev1.Secret{
-			ObjectMeta: metav1.ObjectMeta{
-				Name:      secretName,
-				Namespace: provider.Namespace,
-			},
-			Type: secret.Type,
-		}
+	for secretName, secret := range processor.Settings.SecretSettings {
+		if processor.ShouldReconcile || shouldCreateOrUpdate(reconciler.ProvidersReconcileState[namespacedName].ExistingSecretReferences, secretName, processor.Settings.SecretReferences[secretName]) {
```

Comment:
Nit: I know there's much difference in logic, but I think doing this is better for readability.

1. check `shouldReconcile` in the  `shouldCreateOrUpdate` method

2.
```
if !shouldCreateOrUpdate {
    // do something
  continue
}

// do something
```